src/store/tests/Bridge/Supabase/StoreTest.php

```diff
@@ -0,0 +1,241 @@
+        $uuid1 = Uuid::v4();
+        $uuid2 = Uuid::v4();
+        $expectedResponse = [
+            [
+                'id' => $uuid1->toRfc4122(),
+                'embedding' => '[0.1, 0.2]',
+                'metadata' => '{"type": "first"}',
+                'score' => 0.95,
+            ],
+            [
+                'id' => $uuid2->toRfc4122(),
+                'embedding' => '[0.3, 0.4]',
+                'metadata' => '{"type": "second"}',
+                'score' => 0.85,
+            ],
+        ];
+        $httpClient = new MockHttpClient(new JsonMockResponse($expectedResponse));
+        $store = $this->createStore($httpClient, 2);
+        $queryVector = new Vector([1.0, 2.0]);
+
+        $result = $store->query($queryVector, ['max_items' => 2, 'min_score' => 0.8]);
+
+        $this->assertCount(2, $result);
+        $this->assertInstanceOf(VectorDocument::class, $result[0]);
+        $this->assertTrue($uuid1->equals($result[0]->id));
+        $this->assertSame([0.1, 0.2], $result[0]->vector->getData());
+        $this->assertSame(0.95, $result[0]->score);
+        $this->assertSame(['type' => 'first'], $result[0]->metadata->getArrayCopy());
+        $this->assertInstanceOf(VectorDocument::class, $result[1]);
+        $this->assertTrue($uuid2->equals($result[1]->id));
+        $this->assertSame([0.3, 0.4], $result[1]->vector->getData());
+        $this->assertSame(0.85, $result[1]->score);
+        $this->assertSame(['type' => 'second'], $result[1]->metadata->getArrayCopy());
+
+        $this->assertSame(1, $httpClient->getRequestsCount());
+    }
+
+    public function testQueryParsesComplexMetadata()
+    {
+        $uuid = Uuid::v4();
+        $expectedResponse = [
+            [
+                'id' => $uuid->toRfc4122(),
+                'embedding' => '[0.1, 0.2, 0.3, 0.4]',
+                'metadata' => '{"title": "Test Document", "tags": ["ai", "test"], "score": 0.92}',
+                'score' => 0.92,
+            ],
+        ];
+        $httpClient = new MockHttpClient(new JsonMockResponse($expectedResponse));
+        $store = $this->createStore($httpClient, 3);
+        $queryVector = new Vector([1.0, 2.0, 3.0]);
+
+        $result = $store->query($queryVector);
+
+        $document = $result[0];
+        $metadata = $document->metadata->getArrayCopy();
+        $this->assertCount(1, $result);
+        $this->assertInstanceOf(VectorDocument::class, $document);
+        $this->assertTrue($uuid->equals($document->id));
+        $this->assertSame([0.1, 0.2, 0.3, 0.4], $document->vector->getData());
+        $this->assertSame(0.92, $document->score);
+        $this->assertSame('Test Document', $metadata['title']);
+        $this->assertSame(['ai', 'test'], $metadata['tags']);
+        $this->assertSame(0.92, $metadata['score']);
+        $this->assertSame(1, $httpClient->getRequestsCount());
+    }
+
+    private function createStore(?MockHttpClient $httpClient = null, ?int $vectorDimension = 2): SupabaseStore
```

Comment:
```suggestion
    private function createStore(MockHttpClient $httpClient, ?int $vectorDimension = 2): SupabaseStore
```
We always provide one